lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,712 @@
+    print(f"Imported and registered the following callbacks: {registered_list}")
+
+
+# %%
+# Load the `FinetuningScheduler` PyTorch Lightning extension module we want to use. This will import all necessary callbacks.
+mock_register_module("finetuningscheduler")
+# set notebook-level variables
+AVAIL_GPUS = torch.cuda.device_count()
+TASK_NUM_LABELS = {"boolq": 2, "rte": 2}
+DEFAULT_TASK = "rte"
+
+transformers_logging.set_verbosity_error()
+# ignore warnings related tokenizers_parallelism/DataLoader parallelism trade-off and
+# expected logging behavior
+for warnf in [".*does not have many workers*", ".*The number of training samples.*"]:
+    warnings.filterwarnings("ignore", warnf)
+
+
+# %%
+class RteBoolqDataModule(pl.LightningDataModule):
+    """A ``LightningDataModule`` designed for both the RTE or BoolQ SuperGLUE Hugging Face datasets."""
+
+    TASK_TEXT_FIELD_MAP = {"rte": ("premise", "hypothesis"), "boolq": ("question", "passage")}
+    LOADER_COLUMNS = (
+        "datasets_idx",
+        "input_ids",
+        "token_type_ids",
+        "attention_mask",
+        "start_positions",
+        "end_positions",
+        "labels",
+    )
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        task_name: str = DEFAULT_TASK,
+        max_seq_length: int = 128,
+        train_batch_size: int = 16,
+        eval_batch_size: int = 16,
+        tokenizers_parallelism: bool = True,
+        **dataloader_kwargs: Any,
+    ):
+        """Initialize this ``LightningDataModule`` designed for both the RTE or BoolQ SuperGLUE Hugging Face
+        datasets.
+
+        Args:
+            model_name_or_path (str):
+                Can be either:
+                    - A string, the ``model id`` of a pretrained model hosted inside a model repo on huggingface.co.
+                        Valid model ids can be located at the root-level, like ``bert-base-uncased``, or namespaced under
+                        a user or organization name, like ``dbmdz/bert-base-german-cased``.
+                    - A path to a ``directory`` containing model weights saved using
+                        :meth:`~transformers.PreTrainedModel.save_pretrained`, e.g., ``./my_model_directory/``.
+            task_name (str, optional): Name of the SuperGLUE task to execute. This module supports 'rte' or 'boolq'.
+                Defaults to DEFAULT_TASK which is 'rte'.
+            max_seq_length (int, optional): Length to which we will pad sequences or truncate input. Defaults to 128.
+            train_batch_size (int, optional): Training batch size. Defaults to 16.
+            eval_batch_size (int, optional): Batch size to use for validation and testing splits. Defaults to 16.
+            tokenizers_parallelism (bool, optional): Whether to use parallelism in the tokenizer. Defaults to True.
+        """
+        super().__init__()
+        self.model_name_or_path = model_name_or_path
```

Comment:
let's use `self.save_hyperparameters()`
then you can access them using `self.hparams.argument`.
helps you remove the explicit assignments. Also then they will be saved in the checkpoint too.

Reply:
Good idea! Done.